cpp/include/raft/stats/contingency_matrix.cuh

```diff
@@ -100,6 +148,52 @@ void contingencyMatrix(const T* groundTruth,
                                      maxLabel);
 }
 
+/**
+ * @brief contruct contingency matrix given input ground truth and prediction
+ *        labels. Users should call function getInputClassCardinality to find
+ *        and allocate memory for output. Similarly workspace requirements
+ *        should be checked using function getContingencyMatrixWorkspaceSize
+ * @tparam T label type
+ * @tparam OutT output matrix type
+ * @tparam IdxType Index type of matrix extent.
+ * @tparam LayoutPolicy Layout type of the input data.
+ * @tparam AccessorPolicy Accessor for the input and output, must be valid accessor on
+ *                        device.
+ * @param handle: the raft handle.
+ * @param groundTruth: device 1-d array for ground truth (num of rows)
+ * @param predictedLabel: device 1-d array for prediction (num of columns)
+ * @param outMat: output buffer for contingecy matrix
+ * @param workspace: Optional, workspace memory allocation
+ * @param workspaceSize: Optional, size of workspace memory
+ * @param minLabel: Optional, min value in input ground truth array
+ * @param maxLabel: Optional, max value in input ground truth array
+ */
+template <typename T,
+          typename OutT = int,
+          typename IdxType,
+          typename LayoutPolicy,
+          typename AccessorPolicy>
+void contingencyMatrix(
+  const raft::handle_t& handle,
+  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> groundTruth,
+  raft::mdspan<T, raft::vector_extent<IdxType>, LayoutPolicy, AccessorPolicy> predictedLabel,
+  raft::mdspan<OutT, raft::matrix_extent<IdxType>, LayoutPolicy, AccessorPolicy> outMat,
+  void* workspace      = nullptr,
```

Comment:
If the `workspace` will always be treated as an array of `T` (or `OutT`), then would you consider giving it that appropriate type (at least `T*`, preferably an `mdspan` of `T`)?

Otherwise, if the `workspace` is just a bunch of contiguous bytes, then would you consider making it an mdspan or span of `char` instead of a `void*`?

Using mdspan or span here would let you get rid of the `workspaceSize` parameter.